src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -12,12 +12,54 @@
 using AutoRest.CSharp.Input;
 using AutoRest.CSharp.Mgmt.AutoRest;
 using AutoRest.CSharp.Mgmt.Models;
+using AutoRest.CSharp.Output.Builders;
 using AutoRest.CSharp.Utilities;
 
 namespace AutoRest.CSharp.Mgmt.Decorator
 {
     internal static class CodeModelExtension
     {
+        private static readonly IEnumerable<string> EnumValuesShouldBePrompted = new[]
+        {
+            "None", "NotSet", "Unknown", "NotSpecified", "Unspecified", "Undefined"
+        };
+
+        public static void UpdateSealChoiceTypes(this IEnumerable<Schema> allSchemas)
+        {
+            foreach (var schema in allSchemas)
+            {
+                if (schema is not SealedChoiceSchema choiceSchema)
+                    continue;
+
+                // rearrange the sequence in the choices
+                choiceSchema.Choices = RearrangeChoices(choiceSchema.Choices);
+            }
+        }
+
+        private static ICollection<ChoiceValue> RearrangeChoices(ICollection<ChoiceValue> originalValues)
+        {
+            IEnumerable<ChoiceValue> whateverLeft = originalValues;
+            var result = new List<ChoiceValue>();
+
+            var words = EnumValuesShouldBePrompted.Concat(MgmtContext.MgmtConfiguration.PromptedEnumValues);
+
+            static Func<ChoiceValue, bool> GetFilter(string word)
```

Comment:
I wonder if this is simpler / more efficient to get the index of the one we want to move.  (this assumes we only ever want to move one which I think we need to assert) Then do a RemoveAt(index) and InsertAt(0, item)

Reply:
I kind of prefer to keep this. The case is that we might have multiple entries in an enum that falls into the list, if we change this into some indice manipulations, it might be more complicated.
And since the iterations are constructing by linq, those will only execute once when we are iterating on it (the `ToArray` invocation). Therefore I believe there is no performance impact here.